server/mongo_data_access.go

```diff
@@ -399,8 +399,10 @@ func (dal *mongoDataAccessLayer) Search(baseURL url.URL, searchQuery search.Quer
 	var err error
 	usesIncludes := len(searchQuery.Options().Include) > 0
 	usesRevIncludes := len(searchQuery.Options().RevInclude) > 0
+	usesChainedSearch := testUsesChainedSearch(searchQuery)
+
 	// Only use (slower) pipeline if it is needed
-	if usesIncludes || usesRevIncludes {
+	if usesIncludes || usesRevIncludes || usesChainedSearch {
 		result = models.NewSlicePlusForResourceName(searchQuery.Resource, 0, 0)
 		err = searcher.CreatePipeline(searchQuery).All(result)
```

Comment:
Check if `usesIncludes` or `usesRevIncludes` is true.  If neither are true then just get a slice of the resources rather than a slice of the resource "pluses".